src/algebra/homology/local_cohomology.lean

```diff
@@ -186,4 +194,94 @@ begin
        ... ≤ J           : hk,
 end
 
+/-- The diagram of powers of `J` is initial in the diagram of all ideals with
+radical containing `J`. This uses noetherianness. -/
+/- Can this proof be golfed? -/
+instance ideal_powers_initial [hR : is_noetherian R R] :
+  functor.initial (ideal_powers_to_self_le_radical J) :=
+{ out := λ J', begin
+  apply @zigzag_is_connected _ _ _,
+  { intros j1 j2,
+    apply relation.refl_trans_gen.single,
+    -- The inclusions `J^n1 ≤ J'` and `J^n2 ≤ J'` always form a triangle, based on
+    -- which exponent is larger.
+    cases le_total (unop j1.left) (unop j2.left) with h,
+      right, exact ⟨costructured_arrow.hom_mk (hom_of_le h).op (of_as_true trivial)⟩,
+      left, exact ⟨costructured_arrow.hom_mk (hom_of_le h).op (of_as_true trivial)⟩ },
+  { obtain ⟨k, hk⟩ := ideal.exists_pow_le_of_le_radical_of_fg _ _ J'.2
+      (is_noetherian_def.mp hR _),
+    refine ⟨costructured_arrow.mk _⟩,
+    exact (op k),
+    exact ⟨⟨hk⟩⟩ },
+  end }
```

Comment:
@emwitt @samvang @semorrison I had to write some `structured_arrow`s by hand here.

Reply:
Maybe this can be golfed?

Reply:
It would still be nice to golf this proof.

Reply:
What is "golfing" in this context?

Reply:
It basically means making the proof nicer or tidier. In this case, I'm not sure if there's some API lemmas I was overlooking for making `structured_arrow`s. All the `⟨stuff in brackets⟩` notation is basically constructing objects by hand.

Reply:
In particular, it seemed like there should have been a lemma somewhere to say that `a ≤ b` in a poset should give `a ⟶ b` in this structured category, but I couldn't find one.